src/components/slide-toggle/slide-toggle.scss

```diff
@@ -12,12 +12,26 @@ $md-slide-toggle-thumb-size: 20px !default;
 $md-slide-toggle-margin: 16px !default;
 
 @mixin md-switch-checked($palette) {
-  .md-slide-toggle-thumb {
-    background-color: md-color($palette);
+  &.md-checked {
+    .md-slide-toggle-thumb {
+      background-color: md-color($palette);
+    }
+
+    .md-slide-toggle-bar {
+      background-color: md-color($palette, 0.5);
+    }
   }
+}
+
+@mixin md-switch-ripple($palette) {
+  // Temporary ripple effect for the thumb of the slide-toggle.
+  // Bind to the parent selector and specify the current palette.
+  @include md-temporary-ink-ripple(slide-toggle, true, $palette);
 
-  .md-slide-toggle-bar {
-    background-color: md-color($palette, 0.5);
+  &.md-slide-toggle-focused {
+    &:not(.md-checked) .md-ink-ripple {
+      background-color: rgba(black, 0.12);

```

Comment:
You're right. It should be white on dark themes. This is also something which needs to be addressed in Material 1 (will open an issue)